beta/nncf/tensorflow/graph/converter.py

```diff
@@ -190,3 +143,143 @@ def get_graph_to_layer_var_names_map(concrete_fun):
+
+    return raw_nodes
+
+
+def _get_nncf_graph_from_raw_nodes(model_config: dict, raw_nodes: Dict) -> NNCFGraph:
+    nncf_graph = NNCFGraph()
+    nncf_graph = _update_graph_with_raw_nodes(nncf_graph, raw_nodes, model_config)
+    return nncf_graph
+
+
+def _update_graph_with_raw_nodes(graph: Union[nx.DiGraph, NNCFGraph],
+                                 raw_nodes: Dict,
+                                 model_config: dict) -> Union[nx.DiGraph, NNCFGraph]:
+    for original_name, instances in raw_nodes.items():
+        for i, attributes in instances.items():
+            node_name = get_expanded_node_name(original_name, i, attributes['is_shared'])
+            graph.add_node(node_name, **attributes)
+
+    for layer in model_config['layers']:
+        layer_name = layer['name']
+        for i, inbound_nodes in enumerate(layer['inbound_nodes']):
+            is_shared = raw_nodes[layer_name][i]['is_shared']
+            node_full_name = get_expanded_node_name(layer_name, i, is_shared=is_shared)
+            for producer_name, producer_instance, *_ in inbound_nodes:
+                is_shared = raw_nodes[producer_name][producer_instance]['is_shared']
+                producer_full_name = get_expanded_node_name(producer_name, producer_instance, is_shared=is_shared)
+                graph.add_edge(producer_full_name, node_full_name)
+
+    return graph
+
+
+def _get_nncf_graph_from_sequential(model: tf.keras.Model) -> NNCFGraph:
+    nncf_graph = NNCFGraph()
+    producer_layer = None
+    model_config = model.get_config()
+    for layer in model_config['layers']:
+        layer_name = layer['config']['name']
+        layer_type = _get_layer_type(layer)
+        layer_dtype = _get_layer_dtype(layer)
+        data_format = layer['config'].get('data_format')
+        attrs = dict(type=layer_type,
+                     dtype=layer_dtype,
+                     data_format=data_format,
+                     in_ports=[0],
+                     out_ports=[0],
+                     is_shared=False)
+        if layer_type in GENERAL_CONV_LAYERS:
+            module_attributes = _get_module_attributes(model.get_layer(layer_name), attrs)
+            attrs.update({NNCFGraph.MODULE_ATTRIBUTES: module_attributes})
+
+        nncf_graph.add_node(layer_name, **attrs)
+        if producer_layer is not None:
+            nncf_graph.add_edge(producer_layer, layer_name)
+        producer_layer = layer_name
+
+    return nncf_graph
+
+
+def _get_module_attributes(layer: tf.keras.layers.Layer, attrs: dict) -> ConvolutionModuleAttributes:
+    channel_axis = -1 if attrs['data_format'] == 'channels_last' else 1
+    return ConvolutionModuleAttributes(layer.trainable,
+                                       layer.get_input_shape_at(0)[channel_axis],
+                                       layer.get_output_shape_at(0)[channel_axis],
+                                       layer.strides[0],
+                                       layer.groups)
```

Comment:
Note - you should really think about grouping these methods into a class.

Reply:
The whole converter.py file is currently in intermediate state, since there are at the same time methods creating nx.Digraph graph (to support parts of code where NNCFGraph is not introduced yet i.e. TF quantization and sparsity) and methods creating NNCFGraph graph. In this PR I just introduced some methods to reduce code duplication. When full transition to NNCFGraph happened this part should be refactored anyway.